spring-boot-project/spring-boot-properties-migrator/src/main/java/org/springframework/boot/context/properties/migrator/PropertiesMigrationReporter.java

```diff
@@ -105,10 +107,50 @@ PropertiesMigrationReport getReport() {
 				result.add(name,
 						new PropertyMigration(configurationProperty, metadata, determineReplacementMetadata(metadata)));
 			}
+			getMatchingPropertiesForMapEntries(name, source, metadata, result);
 		}));
 		return result;
 	}
 
+	private void getMatchingPropertiesForMapEntries(String name, ConfigurationPropertySource source,
+			ConfigurationMetadataProperty metadata, MultiValueMap<String, PropertyMigration> result) {
+		if (source instanceof IterableConfigurationPropertySource)
+			((IterableConfigurationPropertySource) source).iterator().forEachRemaining(configurationPropertyName -> {
+				ConfigurationProperty configurationProperty = source.getConfigurationProperty(configurationPropertyName);
```

Comment:
This should be easier, something like source.getConfigurationProperties() would be nice API to add. Not sure if it belongs on PropertySource though

Reply:
How about using `filter`?

Reply:
@dominiccroce do you want to get another look at it? `filter` takes a predicate and should lets you filter the propertysource with only the keys that are relevant. I haven't tried but I think it's worth exploring.